packages/celotool/src/cmds/account/faucet.ts

```diff
@@ -76,45 +98,49 @@ export const handler = async (argv: FaucetArgv) => {
-      kit.contracts.getReserve(),
-    ])
+    const faucetToken = async (tokenParams: TokenParams) => {
+      if (!tokenParams.amount) {
+        return
+      }
 
-    if (argv.checkzero) {
+      const tokenWrapper = await kit.celoTokens.getWrapper(tokenParams.token as any)
       for (const address of addresses) {
-        // Check this address hasn't already been fauceted and has zero gold
-        if (
-          (argv.gold !== 0 && !(await goldToken.balanceOf(address)).isZero()) ||
-          (argv.dollar !== 0 && !(await stableToken.balanceOf(address)).isZero())
-        ) {
-          console.error(
-            `Unable to faucet ${address} on ${argv.celoEnv}: --checkzero specified, but balance is non-zero`
-          )
-          process.exit(1)
+        if (argv.checkzero) {
+          // Check if address account balance of this token is zero
+          if (!(await tokenWrapper.balanceOf(address)).isZero()) {
+            throw Error(
+              `Unable to faucet ${tokenParams.token} to ${address} on ${argv.celoEnv}: --checkzero specified, but balance is non-zero`
+            )
+          }
         }
-      }
-    }
-
-    for (const address of addresses) {
-      const goldAmount = await convertToContractDecimals(argv.gold, goldToken)
-      const stableTokenAmount = await convertToContractDecimals(argv.dollar, stableToken)
-
-      console.log(
-        `Fauceting ${goldAmount.toFixed()} Gold and ${stableTokenAmount.toFixed()} StableToken to ${address}`
-      )
+        const tokenAmount = await convertToContractDecimals(tokenParams.amount, tokenWrapper)
+        console.log(`Fauceting ${tokenAmount.toFixed()} of ${tokenParams.token} to ${address}`)
 
-      if (!goldAmount.isZero()) {
-        if (await reserve.isSpender(account)) {
-          await reserve.transferGold(address, goldAmount.toFixed()).sendAndWaitForReceipt()
-        } else {
-          await goldToken.transfer(address, goldAmount.toFixed()).sendAndWaitForReceipt()
+        if (tokenParams.token === Token.CELO) {
+          // Special handling for reserve transfer
+          const reserve = await kit.contracts.getReserve()
+          if (await reserve.isSpender(account)) {
+            await reserve.transferGold(address, tokenAmount.toFixed()).sendAndWaitForReceipt()
+            return
+          }
         }
+        await tokenWrapper.transfer(address, tokenAmount.toFixed()).sendAndWaitForReceipt()
+        console.log(`Successfully fauceted ${tokenParams.token}`)
       }
-      if (!stableTokenAmount.isZero()) {
-        await stableToken.transfer(address, stableTokenAmount.toFixed()).sendAndWaitForReceipt()
-      }
+    }
+    // Ensure all faucets attempts are independent of failures and report failures.
+    const failures = (
+      await Promise.all(
+        argv.tokenParams.map(async (tokenParams) => {
+          return faucetToken(tokenParams)
+            .then(() => null)
+            .catch((err) => err)
+        })
```

Comment:
`concurrentMap` is nice for things like this

Reply:
Same as @yorhodes 
Also, this handles more than 1 faucet at the same time, and don't really know if the failure text says which token failed. If is not the case, maybe we should add some data there.

Reply:
thank you!! Added concurrentmap + added specifics about the token!